---
title: "Analysis MiRNA-146a Agogo 1998 & 2000"
author: "Welmoed van Loon"
output: pdf_document
---

# Prepare session

```{r setup, echo = T, warning = F, message = F}

knitr::opts_chunk$set(echo = T, warning = F)

library(genetics)
library(HardyWeinberg)
library(BSDA)
library(data.table)
library(rcompanion)
library(DescTools)
library(MASS)
library(kableExtra)
library(tidyverse)
  
# Load data:

setwd("C:/Users/vanloonw/Desktop/malaria_mirna_146")

df_1998_raw <- fread("Agogo-Master 98.csv", header = T, na.strings = c("", "NA")) 
# White space " " will be read as NA allready

df_2000_raw <- fread("A2000-Primi-MASTER%2826 MAY 2014%29.csv", header = T, na.strings = c("", "NA")) 
# White space " " will be read as NA allready

```
\pagebreak

# Prepare dataframes for analysis

## Agogo 1998

```{r prepare data 1998}

## Prepare dataset 1998:

df_1998_all <- df_1998_raw %>%
  select(c(id = `No.`,
           age, 
           loc, 
           grav = `No.-pr.`, 
           grav3 = Grav4,
           gestation = `gest.age`, 
           anc_visits = `No.visits`, 
           pyr = `PYR.2`, 
           clq = `CLQ-num`,
           # Clinic:
           temp = Temp, 
           fever = FEVER,
           hb = HemoCue,
           anaemia = ANA,
           crp = CRP,
           ferritin = Ferritin,
           ferritin_def = `Fe-Mangel`,
           wbc = WBC,
           wbc2 = WBC2,
           rbc = RBC,
           # RBC characteristics:
           iron_duffy = `Eisen(Duffy)`,
           hypochromia = Hypochromia, 
           macrocytosis = macrocytose, 
           microcytosis = MIKROCYTOSIS, 
           # Genotypes/SNPs:
           defi = DEFI,
           a_thal = THAL,
           hb_type = `Hb-Type.3`,
           g6pd = GD, 
           g6pd_def = `Gd-def`,
           fpn_q248h = `FPN Q248H`,
           mirna146 = miRNA,
           # Malaria / parasitemia:
           pcr_falc = FALC,
           pcr_mala = MAL,
           pcr_oval = OVAL,
           microsc = `Micro-infected`,
           pardens = `P/mcl`,
           log_pardens = logpara)) %>%
  
          # note:
          # columns 'micro-pos' and 'Micro-infected' do not
          # fully correspond: ID 167 had different value 
          # for both, which one to take?
  
  mutate(study = rep("anc"), # this makes later selections easier in functions
         primi_multi = ifelse(grav == 1, "primi", 
                              ifelse(grav > 1, "multi", NA)),

         pcr_falc = ifelse(pcr_falc == "+", 1, ifelse(pcr_falc == "-", 0, NA)),
         pcr_mala = ifelse(pcr_mala == "+", 1, ifelse(pcr_mala == "-", 0, NA)),
         pcr_oval = ifelse(pcr_oval == "+", 1, ifelse(pcr_oval == "-", 0, NA)),
         
         pcr_any = ifelse(pcr_falc == 1 | pcr_mala == 1 | pcr_oval == 1, 1, 0),

         microsc = ifelse(microsc == "b: M-positiv", 1, 
                            ifelse(microsc == "a: M-negativ", 0, NA)),
         
         malaria = ifelse(pcr_any == 1 | microsc == 1, 1, 0),
         
         mirna146 = ifelse(mirna146 == "a wt", "a_wt", 
                           ifelse(mirna146 == "b het", "b_het", 
                                  ifelse(mirna146 == "c hom", "c_hom", NA))),
         mirna146_wt_vs_hetorhom = ifelse(mirna146 %in% c("b_het", "c_hom"), 
                                          "b_het_or_hom", mirna146),
         mirna146_wt_vs_hom = ifelse(mirna146 == "b_het", NA, mirna146),
         mirna146_wt_vs_het = ifelse(mirna146 == "c_hom", NA, mirna146),
         mirna146_het_vs_hom = ifelse(mirna146 == "a_wt", NA, mirna146),

         mirna146_allele = str_replace_all(mirna146, c("a_wt" = "G/G", "b_het" = "G/C", "c_hom" = "C/C"))) %>%
  
  # EXCLUDE OBSERVATIONS WITHOUT INFO ABOUT miRNA146 !!
  filter(!is.na(mirna146))


# make sub dataframes with the primigravidae or multigravidea:
df_1998_prim <- df_1998_all %>%
  filter(grav == 1)

df_1998_multi <- df_1998_all %>%
  filter(grav > 1)

```
\pagebreak

## Agogo 2000

```{r prepare data 2000}

## Prepare dataset 2000:

df_2000_all <- df_2000_raw %>%
  filter(`twin/tod` == "ok") %>% 
  
  select(id = M, 
         age = Age, 
         loc = Residence, 
         edu = Edu, 
         grav = Parity,
         anc_visits = `No.visits  ANC`,
         pyr = `PYR-pos`,
         clq = `CQ<5`,
         season = Season, # which season-column to take? 71 mismatches!!
         # Clinic:
         temp = Temp,
         fever = `Fever-M`,
         hb = HemoCue,
         anaemia = `M-ANA`,
         wbc = WBC,
         rbc = RBC,
         lbw = LBW,
         preterm = PREMATURE,
         fever_child = `Fever-C`,
         hb_child = `ch-Hb`,
         anaemia_child = `NEO-ANA`,
         wbc_child = `ch-WBC`,
         rbc_child = `ch-RBC`,
         # RBC characterisics:
         # ...
         # Genotypes:
         g6pd = G6PD,
         Hb_type = Hb3,
         a_thal = `a-Thal2`,
         mirna146 = `miRNA 146`,
         # Malaria / parasitemia:
         pcr_falc_m = `PCR-Fal-M`,
         pcr_falc_p = `PCR-Fal-P`,
         pcr_mala_m = `PCR-Mal-M`,
         pcr_mala_p = `PCR-Mal-P`,
         pcr_oval_m = `PCR-Ova-M`,
         pcr_oval_p = `PCR-Ova-P`,           
         microsc_m = `M.Micro.pos-Uli`,
         microsc_p = `Mikroskopie Pl.`,
         pardens = `P/mcl-M`,
         log_pardens_m = LOGPD,
         log_pardens_p = `LOGPD(plaz)`,
         pigment = `Leuk-Pigment`,
         hrp2_m = `ICT-M`,
         hrp2_p = plac.ICT,
         malaria_p_pastorprsnt = `Any positive Placenta`) %>%
  
  mutate(study = rep("delivery"),
         
         pcr_falc_m = ifelse(pcr_falc_m == "+", 1, ifelse(pcr_falc_m == "-", 0, NA)),
         pcr_mala_m = ifelse(pcr_mala_m == "+", 1, ifelse(pcr_mala_m == "-", 0, NA)),
         pcr_oval_m = ifelse(pcr_oval_m == "+", 1, ifelse(pcr_oval_m == "-", 0, NA)),
         
         pcr_falc_p = ifelse(pcr_falc_p == "+", 1, ifelse(pcr_falc_p == "-", 0, NA)),
         pcr_mala_p = ifelse(pcr_mala_p == "+", 1, ifelse(pcr_mala_p == "-", 0, NA)),
         pcr_oval_p = ifelse(pcr_oval_p == "+", 1, ifelse(pcr_oval_p == "-", 0, NA)),
         
         pcr_falc = ifelse(pcr_falc_m == 1 | pcr_falc_p == 1, 1, 0),
         pcr_mala = ifelse(pcr_mala_m == 1 | pcr_mala_p == 1, 1, 0),
         pcr_oval = ifelse(pcr_oval_m == 1 | pcr_oval_p == 1, 1, 0),
         
         pcr_any_m = ifelse(pcr_falc_m == 1 | pcr_mala_m == 1 | pcr_oval_m == 1, 1, 0),
         pcr_any_p = ifelse(pcr_falc_p == 1 | pcr_mala_p == 1 | pcr_oval_p == 1, 1, 0),
         pcr_any = ifelse(pcr_any_m == 1 | pcr_any_p == 1, 1, 0),
         
         microsc_m = ifelse(microsc_m == "b-Micro-M-positiv", 1, 
                              ifelse(microsc_m == "a-Micro-M-negativ", 0, NA)),
         microsc_p = ifelse(microsc_p == "b-Micro-Pl-pos.", 1, 
                              ifelse(microsc_p == "a-Micro-Pl-neg.", 0, NA)),
         microsc = ifelse(microsc_m == 1 | microsc_p == 1, 1, 0),
         
         hrp2_m = ifelse(hrp2_m == "+", 1, ifelse(hrp2_m == "-", 0, NA)),
         hrp2_p = ifelse(hrp2_p == "+", 1, ifelse(hrp2_p == "-", 0, NA)),
         hrp2 = ifelse(hrp2_m == 1 | hrp2_p == 1, 1, 0),

         pigment = ifelse(pigment == "+", 1, ifelse(pigment == "-", 0, NA)),
         
         malaria_m = ifelse(pcr_any_m == 1 | microsc_m == 1, 1, 0),
         # the defenition of "past or present malaria" 
         # can be slightly different depending on inlcuding hrp2...
         # because of earlier publications, we use the existing column "Any positive Placenta" 
         malaria_p = ifelse(pcr_any_p == 1 | microsc_p == 1 | hrp2_p == 1 | pigment == 1, 1, 0),
         malaria_p_pastorprsnt = as.numeric(str_replace_all(malaria_p_pastorprsnt, c("b pos" = "1", "a - neg" = "0"))),
         malaria = ifelse(malaria_m == 1 | malaria_p == 1 | hrp2_m == 1, 1, 0),
         
         mirna146 = ifelse(mirna146 == "a - wt", "a_wt", 
                           ifelse(mirna146 == "b - het", "b_het",
                                   ifelse(mirna146 == "c - hom", "c_hom", NA))),
         mirna146_wt_vs_hetorhom = ifelse(mirna146 %in% c("b_het", "c_hom"), 
                                          "b_het_or_hom", mirna146),
         mirna146_wt_vs_hom = ifelse(mirna146 == "b_het", NA, mirna146),
         mirna146_wt_vs_het = ifelse(mirna146 == "c_hom", NA, mirna146),
         mirna146_het_vs_hom = ifelse(mirna146 == "a_wt", NA, mirna146),
         
         mirna146_allele = str_replace_all(mirna146, c("a_wt" = "G/G", "b_het" = "G/C", "c_hom" = "C/C"))) %>%

  
  # EXCLUDE OBSERVATIONS WITHOUT miRNA146 data !!
  filter(!is.na(mirna146))
  
```
\pagebreak

# Descriptive statistics

## Functions

```{r}

func_summ <- function(data){
  
  freq_mirna146 <- data %>%
    select(mirna146) %>%
    table() %>%
    addmargins() %>%
    kable()
  
  prop_mirna146 <- data %>%
    select(mirna146) %>%
    table() %>%
    prop.table() %>%
    kable

  df_base_temp <- data %>%
    summarize(n = n(),
              
              age_n    = sum(!is.na(age)),
              age_mean = mean(age, na.rm = T),
              age_sd   = sd(age, na.rm = T),
              age_median = median(age, na.rm = T),
              age_min    = min(age, na.rm = T), 
              age_max    = max(age, na.rm = T),
              
              anc_visits_n    = sum(!is.na(anc_visits)),
              anc_visits_mean = mean(anc_visits, na.rm = T),
              anc_visits_sd   = sd(anc_visits, na.rm = T),
              anc_visits_median = median(anc_visits, na.rm = T),
              anc_visits_min    = min(anc_visits, na.rm = T), 
              anc_visits_max    = max(anc_visits, na.rm = T),
              
              fever_n   = sum(!is.na(fever)),
              fever_yes = sum(fever %in% c("FEVER", "b-fever")),
              fever_perc= sum(fever %in% c("FEVER", "b-fever")) / 
                sum(!is.na(fever)) * 100,
              
              temp_n    = sum(!is.na(temp)),
              temp_mean = mean(temp, na.rm = T),
              temp_sd   = sd(temp, na.rm = T),
              temp_median = median(temp, na.rm = T),
              temp_min    = min(temp, na.rm = T), 
              temp_max    = max(temp, na.rm = T),
              
              anaemia_n   = sum(!is.na(anaemia)),
              anaemia_yes = sum(anaemia %in% c("A<11", "b.ANA")),
              amaemia_perc= sum(anaemia %in% c("A<11", "b.ANA") /
                sum(!is.na(anaemia)) ) * 100,
              
              hb_n = sum(!is.na(hb)),
              hb_mean = mean(hb, na.rm = T),
              hb_sd   = sd(hb, na.rm = T),
              hb_median = median(hb, na.rm = T),
              hb_min    = min(hb, na.rm = T), 
              hb_max    = max(hb, na.rm = T),
  
              pyr_n   = sum(!is.na(pyr)),
              pyr_yes = sum(pyr %in% c("c: pyr-pos", "b-positiv")),
              pyr_perc= sum(pyr %in% c("c: pyr-pos", "b-positiv")) / 
                sum(!is.na(pyr)) * 100) %>%
    
    mutate_if(is.numeric, round, digits = 2) %>%
    t() %>% 
    as.data.frame() %>%
    rownames_to_column()
  
  if("anc" %in% data$study){ # if it comprises the ANC-study
    
      df_base_1998 <- data %>%
        summarize(primigr_n    = sum(grav == 1, na.rm = T),
                  primigr_perc = sum(grav == 1, na.rm = T) / n() * 100,
                  multigr_n    = sum(grav > 1, na.rm = T),
                  multigr_perc = sum(grav > 1, na.rm = T) / n() * 100) %>%
                  
        mutate_if(is.numeric, round, digits = 2) %>%
        t() %>% 
        as.data.frame() %>% 
        rownames_to_column()
      
      df_base <- df_base_temp %>%
        rbind(df_base_1998)
  }
    
  if("delivery" %in% data$study){ # if it comprises the ANC-study
    
      df_base_2000 <- data %>%
        summarize(preterm_n   = sum(!is.na(preterm)),
                  preterm_yes = sum(preterm == "2-premature", na.rm = T),
                  preterm_perc= sum(preterm == "2-premature", na.rm = T) /
                                      sum(!is.na(preterm)) * 100,
                  
                  lbw_n   = sum(!is.na(lbw)),
                  lbw_yes = sum(lbw == "2-LBW", na.rm = T),
                  lbw_perc= sum(lbw == "2-LBW", na.rm = T) / 
                                  sum(!is.na(lbw)) * 100) %>%
                  
        mutate_if(is.numeric, round, digits = 2) %>%
        t() %>% 
        as.data.frame() %>% 
        rownames_to_column()
      
      df_base <- df_base_temp %>%
        rbind(df_base_2000)
  }
  
  df_base <- df_base %>%
    kable()
  
 plot_age <- ggplot(data) +
  geom_histogram(aes(age))

 plot_hb <- ggplot(data) +
  geom_histogram(aes(hb))
  
  if("anc" %in% data$study){
    df_gmpd <- data %>%
      summarize(mean = mean(log_pardens, na.rm = T),
                sd   = sd(log_pardens, na.rm = T),
                n    = n()) %>%
      mutate(which = rep("maternal"),
             gmpd  = round(10^mean),
             CI_1  = round(10^(mean-1.96*(sd/sqrt(n)))),
             CI_2  = round(10^(mean+1.96*(sd/sqrt(n)))) ) %>%
      select(which, n, mean, sd, gmpd, CI_1, CI_2)
      as.data.frame
  } 
  
  if("delivery" %in% data$study){
    
    df_gmpd_m <- data %>%
      summarize(mean = mean(log_pardens_m, na.rm = T),
                sd   = sd(log_pardens_m, na.rm = T),
                n    = n()) %>%
      mutate(which = rep("maternal"),
             gmpd  = round(10^mean),
             CI_1  = round(10^(mean-1.96*(sd/sqrt(n)))),
             CI_2  = round(10^(mean+1.96*(sd/sqrt(n)))) ) %>%
      select(which, n, mean, sd, gmpd, CI_1, CI_2)
      as.data.frame
    
    df_gmpd_p <- data %>%
      summarize(mean = mean(log_pardens_p, na.rm = T),
                sd   = sd(log_pardens_p, na.rm = T),
                n    = n()) %>%
      mutate(which = rep("placental"),
             gmpd  = round(10^mean),
             CI_1  = round(10^(mean-1.96*(sd/sqrt(n)))),
             CI_2  = round(10^(mean+1.96*(sd/sqrt(n)))) ) %>%
      select(which, n, mean, sd, gmpd, CI_1, CI_2)
      as.data.frame
    
    df_gmpd <- df_gmpd_m %>%
      rbind(df_gmpd_p)
  }  
  
  df_gmpd <- df_gmpd %>%
    kable()
    
  df_infect_all <- data %>%
    select(contains("pcr"), contains("microsc"), contains("malaria")) %>%
    summarize_all(funs(xxntotal = n(),
                       xxninfe = sum(. == 1, na.rm = T),
                       xxaperc = sum(. == 1, na.rm = T) / n() * 100)) %>%
    select(order(colnames(.))) %>%
    mutate_if(is.numeric, signif, 3) %>%
    t() %>% 
    as.data.frame() %>% 
    rownames_to_column() %>%
    separate(1, c("infection", "summary"), sep = "_xx") %>%
    reshape(idvar = "infection",  timevar = "summary", direction = "wide") %>%
    arrange(infection) %>%
    kable()
  
  df_infect_per_snp_1 <- data %>%
    select(mirna146, contains("pcr"), contains("microsc"), contains("malaria")) %>%
    group_by(mirna146) %>%
    summarize_all(funs(xxntotal = n(),
                       xxninfe = sum(. == 1, na.rm = T),
                       xxaperc = sum(. == 1, na.rm = T) / n() * 100)) %>%
    as.data.frame() %>%
    mutate_if(is.numeric, signif, 3) %>%
    column_to_rownames(var = "mirna146") %>%
    select(order(colnames(.))) %>%
    t() %>% 
    as.data.frame() %>% 
    rownames_to_column() %>%
    separate(1, c("infection", "summary"), sep = "_xx") %>%
    gather(genotype, n, -c(infection, summary)) %>%
    reshape(idvar = c("infection","genotype"), 
            timevar = "summary", direction = "wide") %>%
    arrange(infection, genotype) %>%
    kable()
  
   df_infect_per_snp_2 <- data %>%
     select(mirna146_wt_vs_hetorhom, contains("pcr"), 
            contains("microsc"), contains("malaria")) %>%
     group_by(mirna146_wt_vs_hetorhom) %>%
     summarize_all(funs(xxntotal = n(),
                        xxninfe = sum(. == 1, na.rm = T),
                        xxaperc = sum(. == 1, na.rm = T) / n() * 100)) %>%
     as.data.frame() %>%
     mutate_if(is.numeric, signif, 3) %>%
     column_to_rownames(var = "mirna146_wt_vs_hetorhom") %>%
     select(order(colnames(.))) %>%
     t() %>% 
    as.data.frame() %>% 
    rownames_to_column() %>%
    separate(1, c("infection", "summary"), sep = "_xx") %>%
    gather(genotype, n, -c(infection, summary)) %>%
    reshape(idvar = c("infection","genotype"), 
            timevar = "summary", direction = "wide") %>%
    arrange(infection, genotype) %>%
    kable()
   
   lst_summ <- list(freqency_table_mirna146 = freq_mirna146,
                    proportion_table_mirna146 = prop_mirna146,
                    base = df_base, 
                    plot_age,
                    plot_hb,
                    gmpd = df_gmpd,
                    infection_status_all = df_infect_all,
                    infection_status_per_snp_1 = df_infect_per_snp_1,
                    infection_status_per_snp_2 = df_infect_per_snp_2)
  
  return(lst_summ)
  
}

```
\pagebreak

## Agogo 1998

### Multigravidae + primigravidae

```{r}

data <- df_1998_all
func_summ(data)
```

\pagebreak

### Multigravidae only

```{r}

data <- df_1998_multi
func_summ(data)
```

\pagebreak

### Primigravidae only

```{r}

data <- df_1998_prim
func_summ(data)
```

\pagebreak

## Agogo 2000

```{r}

data <- df_2000_all
func_summ(data)
```

\pagebreak

# Test some variables against infection status

## Funcions

```{r}

#...
```

## Agogo 1998

### Multigravidae + primigravidae

```{r}

tbl <- df_1998_all %>%
  select(pcr_falc, primi_multi) %>%
  table()
tbl
fisher.test(tbl, conf.int = T)

```
\pagebreak

# Hardy-Weinberg Equilibrium

## Functions

```{r}

func_HWE <- function(data){
  
  n_obs <- data %>%
    select(mirna146) %>% 
    table() 
  
  wt <- n_obs %>% .["a_wt"] %>% unname
  ht <- n_obs %>% .["b_het"] %>% unname
  mu <- n_obs %>% .["c_hom"] %>% unname
  
  HW_test <- HWChisq(c(wt, ht, mu), cc = F)
  # cc = contingency correction
  
  lst_results <- list(chisq_value = HW_test$chisq , 
                      p_value = HW_test$pval, 
                      expected = HW_test$expected)

  return(lst_results)
}
  
```

\pagebreak

## Agogo 1998

### Multigravidae + primigravidae

```{r}

data <- df_1998_all
func_HWE(data)

```
\pagebreak

### Multigravidae only

```{r}

data <- df_1998_multi
func_HWE(data)

```
\pagebreak

### Primigravidae only

```{r}

data <- df_1998_prim
func_HWE(data)

```
\pagebreak

## Agogo 2000

```{r}

data <- df_2000_all
func_HWE(data)

```
\pagebreak

# Test allele frequencies

## Functions

```{r}

# Functions to make a vector of the allele frequencies or counts for a subset of the data
func_allelcount <- function(data, varname){
  
  vctr_allelefreq <- data %>%
    .[,varname] %>%
    genotype() %>%
    summary() %>%
    .$allele.freq %>%
    .[,"Count"]
  
  return(vctr_allelefreq)
}


func_allelfreq <- function(data, varname){
  
  vctr_allelefreq <- data %>%
    .[,varname] %>%
    genotype() %>%
    summary() %>%
    .$allele.freq %>%
    .[,"Proportion"]
  
  return(vctr_allelefreq)
}
  
```

## Agogo 1998

### Multigravidae + primigravidae

```{r}

data <- df_1998_all

genotype(data$mirna146_allele) %>% summary() %>% .$allele.freq

# Make table of allele frequencies per infection status:

allelcount_pos <- data %>% 
  filter(pcr_falc == 1) %>%
  func_allelcount(.,"mirna146_allele")

allelcount_neg <- data %>% 
  filter(pcr_falc == 0) %>%
  func_allelcount(.,"mirna146_allele")

tbl_allelcount <- data.frame(pcrneg = allelcount_pos, 
                             pcrpos = allelcount_neg) %>%
  as.matrix() 

# summary:
tbl_allelcount %>% addmargins()
tbl_allelcount %>% prop.table(margin = 2)

# test:
fisher.test(tbl_allelcount) 

```

\pagebreak

### Multigravidae only

```{r}

data <- df_1998_multi

genotype(data$mirna146_allele) %>% summary() %>% .$allele.freq

# Make table of allele frequencies per infection status:

allelcount_pos <- data %>% 
  filter(pcr_falc == 1) %>%
  func_allelcount(.,"mirna146_allele")

allelcount_neg <- data %>% 
  filter(pcr_falc == 0) %>%
  func_allelcount(.,"mirna146_allele")

tbl_allelcount <- data.frame(pcrneg = allelcount_pos, 
                             pcrpos = allelcount_neg) %>%
  as.matrix() 

# summary:
tbl_allelcount %>% addmargins()
tbl_allelcount %>% prop.table(margin = 2)

# test:
fisher.test(tbl_allelcount) 


```

\pagebreak

### Primigravidae only

```{r}

data <- df_1998_prim


genotype(data$mirna146_allele) %>% summary() %>% .$allele.freq

# Make table of allele frequencies per infection status:

allelcount_pos <- data %>% 
  filter(pcr_falc == 1) %>%
  func_allelcount(.,"mirna146_allele")

allelcount_neg <- data %>% 
  filter(pcr_falc == 0) %>%
  func_allelcount(.,"mirna146_allele")

tbl_allelcount <- data.frame(pcrneg = allelcount_pos, 
                             pcrpos = allelcount_neg) %>%
  as.matrix() 

# summary:
tbl_allelcount %>% addmargins()
tbl_allelcount %>% prop.table(margin = 2)

# test:
fisher.test(tbl_allelcount) 


```

\pagebreak

## Agogo 2000

### PCR P. falc. peripheral blood

```{r}

data <- df_2000_all

genotype(data$mirna146_allele) %>% summary() %>% .$allele.freq

# Make table of allele frequencies per infection status:

allelcount_pos <- data %>% 
  filter(pcr_falc == 1) %>%
  func_allelcount(.,"mirna146_allele")

allelcount_neg <- data %>% 
  filter(pcr_falc == 0) %>%
  func_allelcount(.,"mirna146_allele")

tbl_allelcount <- data.frame(pcrneg = allelcount_pos, 
                             pcrpos = allelcount_neg) %>%
  as.matrix()

# summary:
tbl_allelcount %>% addmargins()
tbl_allelcount %>% prop.table(margin = 2)

# test:
fisher.test(tbl_allelcount) 

```

\pagebreak

### PCR P. falc. placental blood

```{r}

data <- df_2000_all

genotype(data$mirna146_allele) %>% summary() %>% .$allele.freq

# Make table of allele frequencies per infection status:

allelcount_pos <- data %>% 
  filter(pcr_falc_p == 1) %>%
  func_allelcount(.,"mirna146_allele")

allelcount_neg <- data %>% 
  filter(pcr_falc_p == 0) %>%
  func_allelcount(.,"mirna146_allele")

tbl_allelcount <- data.frame(pcrneg = allelcount_pos, 
                             pcrpos = allelcount_neg) %>%
  as.matrix()

# summary:
tbl_allelcount %>% addmargins()
tbl_allelcount %>% prop.table(margin = 2)

# test:
fisher.test(tbl_allelcount) 

```

\pagebreak

### Past or present placental malaria

```{r}

genotype(data$mirna146_allele) %>% summary() %>% .$allele.freq

# Make table of allele frequencies per infection status:

allelcount_pos <- data %>% 
  filter(malaria_p_pastorprsnt == 1) %>%
  func_allelcount(.,"mirna146_allele")

allelcount_neg <- data %>% 
  filter(malaria_p_pastorprsnt == 0) %>%
  func_allelcount(.,"mirna146_allele")

tbl_allelcount <- data.frame(pcrneg = allelcount_pos, 
                             pcrpos = allelcount_neg) %>%
  as.matrix()

# summary:
tbl_allelcount %>% addmargins()
tbl_allelcount %>% prop.table(margin = 2)
  
# test:
fisher.test(tbl_allelcount) 

```

\pagebreak

# Fisher's exact test for MiRNA146 * infection

## Functions

```{r}

# Function to execute Fisher test on 2 categorical variables:
func_fisher_cochran <- function(data, var_dep, var_indep){
  
  tbl <- data %>%
    select(!!var_indep, !!var_dep) %>%
    table()
  
  # Fisher & Cochran test only possible 
  # when matrix of min. dim 2x2
  # in case not; use tryCathc() to return NULL
  
  tryCatch(
    fisher <- fisher.test(tbl), error = function(e){NULL}
    )
  
  if(length(unique(na.omit(data[,var_indep]))) > 2){
    tryCatch(
      cochran <- CochranArmitageTest(tbl), error = function(e){NULL}
      )
  }else{
    cochran <- "Not relevant for < 3 categories"
  }
  
  tbl_n <- tbl %>% addmargins()
  tbl_p <- tbl %>% prop.table(margin = 1) %>% 
    addmargins(margin = 2) %>% 
    .[,]*100 
  
  lst <- list("var_dep" = var_dep, "var_indep" = var_indep, 
              "table_count" = tbl_n, "table_freq" = tbl_p,
              "results_fisher_test" = fisher, "results_cochran_test" = cochran)
  
  return(lst)
  
}

# Function to execute fisher test on a variable versus many variables:
func_fisher_cochran_multiple <- function(data, vars_dep, vars_indep){
  
  df_vars_combi <- 
    expand.grid(var_indep = vars_indep, var_dep = vars_dep) %>%
    mutate_all(as.character) # factors were behaving strange in the loop...
  
  lst <- list()
  
  for(i in 1:nrow(df_vars_combi)){
    
    var_dep_i   <- df_vars_combi$var_dep[i]
    var_indep_i <- df_vars_combi$var_indep[i]

    lst[[i]] <- func_fisher_cochran(data, var_dep_i, var_indep_i)
      
  }
  
  return(lst)
}

# Function to select output fisher test for publishing:
func_select_output_fisher <- function(lst_results){
  
  df <- data.frame(matrix(ncol = 6, nrow = 0)) %>%
    setNames(c("infect", "SNPs", "OR", 
               "CI_2.5", "CI_97.5", "P"))
  
  for(i in 1:length(lst_results)){
    
    var_dep   <- lst_results[[i]]$var_dep
    var_indep <- lst_results[[i]]$var_indep
    p         <- lst_results[[i]]$results_fisher_test$p.value
    
    if(is.null(lst_results[[i]]$results_fisher_test$estimate[[1]])){
      or     <- NA
      ci_2.5 <- NA
      ci_97.5 <- NA
    }else{
      or      <- lst_results[[i]]$results_fisher_test$estimate[[1]]
      ci_2.5  <- lst_results[[i]]$results_fisher_test$conf.int[[1]]
      ci_97.5 <- lst_results[[i]]$results_fisher_test$conf.int[[2]]
    }
    
    
    df_1row <- data.frame(infect = var_dep, SNPs = var_indep, 
                          OR = or, CI_2.5 = ci_2.5, 
                          CI_97.5 = ci_97.5, P = p)
    
    df <- rbind(df, df_1row)
  }
  
  df <- df %>%
    mutate_if(is.numeric, signif, 2) %>%
    mutate(OR_CI = paste(OR, " (", CI_2.5, "-", CI_97.5, ")", sep = "")) %>%
    select(infect, SNPs, OR_CI, P) %>%
    kable()
  
  return(df)
}

```
\pagebreak

## Agogo 1998

```{r}

## The variables to test:

vars_dep <- c("pcr_falc", "pcr_any", "microsc", "malaria")
vars_indep <- c("mirna146", "mirna146_wt_vs_het", "mirna146_wt_vs_hom", 
                "mirna146_wt_vs_hetorhom", "mirna146_het_vs_hom")

expand.grid(var_indep = vars_indep, var_dep = vars_dep) %>%
  kable()

```

### Multigravidae + primigravidae

```{r}

data <- df_1998_all

lst_results <- func_fisher_cochran_multiple(data, vars_dep, vars_indep)
func_select_output_fisher(lst_results)
lst_results

```
\pagebreak

### Multigravidae only

```{r}

data <- df_1998_multi

lst_results <- func_fisher_cochran_multiple(data, vars_dep, vars_indep)
func_select_output_fisher(lst_results)
lst_results

```
\pagebreak

### Primigravidae only

```{r}

data <- df_1998_prim

lst_results <- func_fisher_cochran_multiple(data, vars_dep, vars_indep)
func_select_output_fisher(lst_results)
lst_results

```
\pagebreak

## Agogo 2000

```{r}

## The variables to test:

vars_dep <- c("pcr_falc", "pcr_falc_m","pcr_falc_p",
              "pcr_any", "pcr_any_m", "pcr_any_p",
              "microsc",  "microsc_m",  "microsc_p",
              "malaria", "malaria_m", "malaria_p_pastorprsnt")
vars_indep <- c("mirna146", "mirna146_wt_vs_het", "mirna146_wt_vs_hom", 
                "mirna146_wt_vs_hetorhom", "mirna146_het_vs_hom")

expand.grid(var_indep = vars_indep, var_dep = vars_dep) %>%
  kable()

```

```{r}

data <- df_2000_all

lst_results <- func_fisher_cochran_multiple(data, vars_dep, vars_indep)
func_select_output_fisher(lst_results)
lst_results

```
\pagebreak

# Logistic regression models for MiRNA146 * infection

## Functions

```{r}

# Funcitons to build log. regressoin model
# and save pretty output
func_log_regr <- function(data, var_resp, var_pred, year){
  
  # both datasets might need different model
  
  if(year == 1998){
    
    model <- glm(get(var_resp) ~ age + anc_visits + pyr + get(var_pred), 
               data = data, family = "binomial")
  }
  
  if(year == 2000){
    
        model <- glm(get(var_resp) ~ age + anc_visits + pyr + season + get(var_pred), 
               data = data, family = "binomial")
  }
  
  results_model <- summary(model)
  
  df_model <- 
    cbind(
      exp(coef(model)),     # exp to make log odds -> odds
      exp(confint(model)),  # exp to make log odds -> odds
      summary(model)$coefficients[,4]) %>%
    as.data.frame() %>%
    rownames_to_column() %>%
    setNames(c("var_predict","OR", "CI_2.5", "CI_97.5", "P"))

  lst <- list(var_predictor = var_pred,
              var_response = var_resp,
              results_model_raw = results_model,
              df_model_exponent = df_model)  
  
  return(lst)
}

# Function to execute log. regression
# on multiple variable combinations:
func_log_regr_multiple <- function(data, vars_resp, vars_pred, year){
  
  df_vars_combi <- 
    expand.grid(var_pred = vars_pred, var_resp = vars_resp) %>%
    mutate_all(as.character) # factors were behaving strange in the loop...
  
  lst <- list()
  
  for(i in 1:nrow(df_vars_combi)){
    
    var_resp_i <- df_vars_combi$var_resp[i]
    var_pred_i <- df_vars_combi$var_pred[i]
    
    lst[[i]] <- func_log_regr(data, var_resp_i, var_pred_i, year)
    
  } 
  
  return(lst)
}

# Function to select output log.regr. for publishing:
func_select_output_log_regr <- function(lst_results){
  
  df <- data.frame(matrix(ncol = 6, nrow = 0)) %>%
    setNames(c("infect", "SNPs", "aOR", "CI_2.5", "CI_97.5", "P"))
  
  for(i in 1:length(lst_results)){
    
    var_resp <- lst_results[[i]]$var_resp

    df_1model <- lst_results[[i]]$df_model_exponent %>%
      filter(stringr::str_detect(var_predict, "a_wt|b_het|c_hom")) %>%
      mutate(var_response = rep(var_resp)) %>%
      select(infect = var_response, SNPs = var_predict, 
             aOR = OR, CI_2.5, CI_97.5, P)

    df <- rbind(df, df_1model)
  }
  
  df <- df %>%
    mutate_if(is.numeric, signif, 2) %>%
    mutate(aOR_CI = paste(aOR, " (", CI_2.5, "-", CI_97.5, ")", sep = "")) %>%
    select(infect, SNPs, aOR_CI, P) %>%
    kable()
  
  return(df)

}

```

## Agogo 1998

```{r}
year <- 1998

## The variables to test:
vars_resp <- c("pcr_falc", "pcr_any", 
               "microsc", "malaria")
vars_pred <- c("mirna146", "mirna146_wt_vs_hetorhom")

expand.grid(var_indep = vars_indep, var_dep = vars_dep) %>%
  kable()

```


### Multigravidae + primigravidae

```{r}

data <- df_1998_all
lst_results <- func_log_regr_multiple(data, vars_resp, vars_pred, year)
func_select_output_log_regr(lst_results)
lst_results

```
\pagebreak

### Multigravidae only

```{r}

data <- df_1998_multi
lst_results <- func_log_regr_multiple(data, vars_resp, vars_pred, year)
func_select_output_log_regr(lst_results)
lst_results

```
\pagebreak

### Primigravidae only

```{r}

data <- df_1998_prim
lst_results <- func_log_regr_multiple(data, vars_resp, vars_pred, year)
func_select_output_log_regr(lst_results)
lst_results

```
\pagebreak

## Agogo 2000

```{r}

year <- 2000

## The variables to test:

vars_resp <- c("pcr_falc", "pcr_falc_m","pcr_falc_p",
               "pcr_any", "pcr_any_m", "pcr_any_p",
               "microsc",  "microsc_m",  "microsc_p",
               "malaria", "malaria_m", "malaria_p_pastorprsnt")
vars_pred <- c("mirna146", "mirna146_wt_vs_hetorhom")

expand.grid(var_indep = vars_indep, var_dep = vars_dep) %>%
  kable()

```


```{r}

data <- df_2000_all
lst_results <- func_log_regr_multiple(data, vars_resp, vars_pred, year)
func_select_output_log_regr(lst_results)
lst_results

```
\pagebreak

# Fisher's exact test for MiRNA146 * clinical manifestation 

## Functions

```{r}

# See functions in "Fisher's exact test for MiRNA146 * infection"

```
\pagebreak

## All participants

### Agogo 1998

```{r}

## The variables to test:

vars_dep <- c("fever", "anaemia")
vars_indep <- c("mirna146", "mirna146_wt_vs_het", "mirna146_wt_vs_hom", 
                "mirna146_wt_vs_hetorhom", "mirna146_het_vs_hom")

expand.grid(var_indep = vars_indep, var_dep = vars_dep) %>%
  kable()

```

#### Multigravidae + primigravidae

```{r}

data <- df_1998_all

lst_results <- func_fisher_cochran_multiple(data, vars_dep, vars_indep)
func_select_output_fisher(lst_results)
lst_results

```
\pagebreak

#### Multigravidae only

```{r}

data <- df_1998_multi

lst_results <- func_fisher_cochran_multiple(data, vars_dep, vars_indep)
func_select_output_fisher(lst_results)
lst_results

```
\pagebreak

#### Primigravidae only

```{r}

data <- df_1998_prim

lst_results <- func_fisher_cochran_multiple(data, vars_dep, vars_indep)
func_select_output_fisher(lst_results)
lst_results

```
\pagebreak

### Agogo 2000

```{r}

## The variables to test:

vars_dep <- c("fever", "anaemia", "preterm", "lbw")
vars_indep <- c("mirna146", "mirna146_wt_vs_het", "mirna146_wt_vs_hom", 
                "mirna146_wt_vs_hetorhom", "mirna146_het_vs_hom")

expand.grid(var_indep = vars_indep, var_dep = vars_dep) %>%
  kable()

```

```{r}

data <- df_2000_all

lst_results <- func_fisher_cochran_multiple(data, vars_dep, vars_indep)
func_select_output_fisher(lst_results)
lst_results

```
\pagebreak

## Within cases only

### Agogo 1998

```{r}

## The variables to test:

vars_dep <- c("fever", "anaemia")
vars_indep <- c("mirna146", "mirna146_wt_vs_het", "mirna146_wt_vs_hom", 
                "mirna146_wt_vs_hetorhom", "mirna146_het_vs_hom")

expand.grid(var_indep = vars_indep, var_dep = vars_dep) %>%
  kable()

```

#### Multigravidae + primigravidae

```{r}

data <- df_1998_all %>% 
  filter(pcr_falc == 1)

lst_results <- func_fisher_cochran_multiple(data, vars_dep, vars_indep)
func_select_output_fisher(lst_results)
lst_results

```
\pagebreak

#### Multigravidae only

```{r}

data <- df_1998_multi %>% 
  filter(pcr_falc == 1)

lst_results <- func_fisher_cochran_multiple(data, vars_dep, vars_indep)
func_select_output_fisher(lst_results)
lst_results

```
\pagebreak

#### Primigravidae only

```{r}

data <- df_1998_prim %>% 
  filter(pcr_falc == 1)

lst_results <- func_fisher_cochran_multiple(data, vars_dep, vars_indep)
func_select_output_fisher(lst_results)
lst_results

```
\pagebreak

### Agogo 2000

```{r}

## The variables to test:

vars_dep <- c("fever", "anaemia", "preterm", "lbw")
vars_indep <- c("mirna146", "mirna146_wt_vs_het", "mirna146_wt_vs_hom", 
                "mirna146_wt_vs_hetorhom", "mirna146_het_vs_hom")

expand.grid(var_indep = vars_indep, var_dep = vars_dep) %>%
  kable()

```

#### Peripherial - PCR

```{r}

data <- df_2000_all %>% 
  filter(pcr_falc_m == 1)

lst_results <- func_fisher_cochran_multiple(data, vars_dep, vars_indep)
func_select_output_fisher(lst_results)
lst_results

```
\pagebreak

#### Placental - PCR

```{r}

data <- df_2000_all %>% 
  filter(pcr_falc_p == 1)

lst_results <- func_fisher_cochran_multiple(data, vars_dep, vars_indep)
func_select_output_fisher(lst_results)
lst_results

```
\pagebreak

#### Placental - Past or present malaria

```{r}

data <- df_2000_all %>% 
  filter(malaria_p_pastorprsnt == 1)

lst_results <- func_fisher_cochran_multiple(data, vars_dep, vars_indep)
func_select_output_fisher(lst_results)
lst_results

```
\pagebreak

# T-test for MiRNA146 * GMPD 

## Functions
```{r}

# Function to do t-test on GMPD data:
func_ttest_gmpd <- function(data, var_infection, var_logpardens){
  
df_gmpd <- data %>%
  filter(get(var_infection) == 1) %>%
  group_by(mirna146) %>%
  summarize(mean = mean(get(var_logpardens), na.rm = T),
            sd = sd(get(var_logpardens), na.rm = T),
            gmpd = round(10^mean),
            n = n())

comb_vars <- crossing(df_gmpd["mirna146"], df_gmpd["mirna146"]) %>% as.data.frame
comb_mean <- crossing(df_gmpd["mean"], df_gmpd["mean"]) %>% as.data.frame
comb_sd   <- crossing(df_gmpd["sd"], df_gmpd["sd"]) %>% as.data.frame
comb_gmpd <- crossing(df_gmpd["gmpd"], df_gmpd["gmpd"]) %>% as.data.frame
comb_n    <- crossing(df_gmpd["n"], df_gmpd["n"]) %>% as.data.frame

mtrx_restults  <- 
  
  mapply(FUN = tsum.test,
         
         mean.x = comb_mean[,1],
         s.x = comb_sd[,1],
         n.x = comb_n[,1],
         
         mean.y = comb_mean[,2],
         s.y = comb_sd[,2],
         n.y = comb_n[,2])

# extract the p-values of each test result
p_values <- apply(MARGIN = 2, X = mtrx_restults, 
                  FUN = function(x){ x$p.value })

# make a df of all info:
df_gmpd_ttest <- data.frame(cat_1 = comb_vars[,1],
                            cat_2 = comb_vars[,2],
                            p_value = p_values) %>%
  distinct(p_value, .keep_all = T)

lst_gmpd_results <- list(df_gmpd = df_gmpd, df_gmpd_ttest = df_gmpd_ttest)

return(lst_gmpd_results)

}
```

## Agogo 1998

### Multigravidae + primigravidae
```{r}
data <- df_1998_all
var_infection <- "pcr_falc"
var_logpardens <- "log_pardens"

func_ttest_gmpd(data, var_infection, var_logpardens)
```

### Multigravidae only
```{r}
data <- df_1998_multi
var_infection <- "pcr_falc"
var_logpardens <- "log_pardens"

func_ttest_gmpd(data, var_infection, var_logpardens)
```

### Primigravidae only
```{r}
data <- df_1998_prim
var_infection <- "pcr_falc"
var_logpardens <- "log_pardens"

func_ttest_gmpd(data, var_infection, var_logpardens)
```


## Agogo 2000

### Peripherial - PCR
```{r}
data <- df_2000_all
var_infection <- "pcr_falc_m"
var_logpardens <- "log_pardens_m"

func_ttest_gmpd(data, var_infection, var_logpardens)
```

### Placental - PCR
```{r}
data <- df_2000_all
var_infection <- "pcr_falc_p"
var_logpardens <- "log_pardens_p"

func_ttest_gmpd(data, var_infection, var_logpardens)
```

### Placental - Past or present malaria
```{r}
data <- df_2000_all
var_infection <- "malaria_p_pastorprsnt"
var_logpardens <- "log_pardens_p"

func_ttest_gmpd(data, var_infection, var_logpardens)
```

# Anova for MiRNA146 * clinical manifestation 

## Functions
```{r}

# Function to execute ANOVA test on 1 categorical and 1 continuous variable:
func_anova <- function(data, var_dep, var_indep){

  df_summary <- data %>%
    select(!!var_dep, !!var_indep) %>%
    group_by_(var_indep) %>%
    summarize_all(funs(mean = mean(., na.rm = T),
                       median = median(., na.rm = T),
                       sd   = sd(., na.rm = T),
                       min  = min(., na.rm = T),
                       max  = max(., na.rm = T))) %>% 
    as.data.frame() %>%
    kable()
  
  # pass character objects to lm() as formula:
  form <- paste(var_dep, "~", var_indep) %>% as.formula
  # run test, put NULL when test is not possible:
  model <- tryCatch(lm(form, data = data), error = function(e){ NULL })
  
  # store results:
  if(is.null(model)){
    p_value <- NA
  }else{
    p_value <- anova(model)$`Pr(>F)`[1]
  } 
  
  results_anova <- data.frame(p = p_value)
    
  lst <- list("var_dep" = var_dep, "var_indep" = var_indep, 
              "summary" = df_summary, 
              "results_anova" = results_anova)
  
  return(lst)
  
}

# Function to make parameter grid of al variable combinations to test, 
# then store results in list:
func_anova_multiple <- function(data, vars_dep, vars_indep){
  
  df_vars_combi <- 
    expand.grid(var_indep = vars_indep, var_dep = vars_dep) %>%
    mutate_all(as.character) # factors were behaving strange in the loop...
  
  lst <- list()
  
  for(i in 1:nrow(df_vars_combi)){
    
    var_dep_i   <- df_vars_combi$var_dep[i]
    var_indep_i <- df_vars_combi$var_indep[i]
    
    lst[[i]] <- func_anova(data, var_dep_i, var_indep_i)
    
  }
  
  return(lst)
}

# Function to select output ANOVA model for quick overview:
func_select_output_anova <- function(lst_results){
  
  df <- data.frame(matrix(ncol = 3, nrow = 0)) %>%
    setNames(c("clinical_manif", "SNPs", "P"))
  
  for(i in 1:length(lst_results)){
    
    var_dep   <- lst_results[[i]]$var_dep
    var_indep <- lst_results[[i]]$var_indep
    p         <- lst_results[[i]]$results_anova$p

    df_1row <- data.frame(clinical_manif = var_dep, 
                          SNPs = var_indep, P = p)
    
    df <- rbind(df, df_1row)
  }
  
  df <- df %>%
    kable()
  
  return(df)
}

```
\pagebreak

## all participants

### Agogo 1998

```{r}

## The variables to test:

vars_dep <- c("temp", "hb")
vars_indep <- c("mirna146", "mirna146_wt_vs_het", "mirna146_wt_vs_hom", 
                "mirna146_wt_vs_hetorhom", "mirna146_het_vs_hom")

expand.grid(var_indep = vars_indep, var_dep = vars_dep) %>%
  kable()
```
\pagebreak

#### Multigravidae + primigravidae

```{r}

data <- df_1998_all

lst_results <- func_anova_multiple(data, vars_dep, vars_indep)
func_select_output_anova(lst_results)
lst_results
```
\pagebreak

#### Multigravidae only

```{r}

data <- df_1998_multi

lst_results <- func_anova_multiple(data, vars_dep, vars_indep)
func_select_output_anova(lst_results)
lst_results
```
\pagebreak

#### Primigravidae only

```{r}

data <- df_1998_prim

lst_results <- func_anova_multiple(data, vars_dep, vars_indep)
func_select_output_anova(lst_results)
lst_results
```
\pagebreak

### Agogo 2000

```{r}

## The variables to test:

vars_dep <- c("temp", "hb")
vars_indep <- c("mirna146", "mirna146_wt_vs_het", "mirna146_wt_vs_hom", 
                "mirna146_wt_vs_hetorhom", "mirna146_het_vs_hom")

expand.grid(var_indep = vars_indep, var_dep = vars_dep) %>%
  kable()
```
\pagebreak

```{r}

data <- df_2000_all

lst_results <- func_anova_multiple(data, vars_dep, vars_indep)
func_select_output_anova(lst_results)
lst_results
```
\pagebreak

## Within cases only

### Agogo 1998

```{r}

## The variables to test:

vars_dep <- c("temp", "hb")
vars_indep <- c("mirna146", "mirna146_wt_vs_het", "mirna146_wt_vs_hom", 
                "mirna146_wt_vs_hetorhom", "mirna146_het_vs_hom")

expand.grid(var_indep = vars_indep, var_dep = vars_dep) %>%
  kable()
```
\pagebreak

#### Multigravidae + primigravidae

```{r}

data <- df_1998_all %>% 
  filter(pcr_falc == 1)

lst_results <- func_anova_multiple(data, vars_dep, vars_indep)
func_select_output_anova(lst_results)
lst_results
```
\pagebreak

#### Multigravidae only

```{r}

data <- df_1998_multi %>% 
  filter(pcr_falc == 1)

lst_results <- func_anova_multiple(data, vars_dep, vars_indep)
func_select_output_anova(lst_results)
lst_results
```
\pagebreak

#### Primigravidae only

```{r}

data <- df_1998_prim %>% 
  filter(pcr_falc == 1)

lst_results <- func_anova_multiple(data, vars_dep, vars_indep)
func_select_output_anova(lst_results)
lst_results
```
\pagebreak

### Agogo 2000

```{r}

## The variables to test:

vars_dep <- c("temp", "hb")
vars_indep <- c("mirna146", "mirna146_wt_vs_het", "mirna146_wt_vs_hom", 
                "mirna146_wt_vs_hetorhom", "mirna146_het_vs_hom")

expand.grid(var_indep = vars_indep, var_dep = vars_dep) %>%
  kable()
```
\pagebreak

#### Peripherial - PCR

```{r}

data <- df_2000_all %>% 
  filter(pcr_falc_m == 1)

lst_results <- func_anova_multiple(data, vars_dep, vars_indep)
func_select_output_anova(lst_results)
lst_results
```
\pagebreak

#### Placental - PCR

```{r}

data <- df_2000_all %>% 
  filter(pcr_falc_p == 1)

lst_results <- func_anova_multiple(data, vars_dep, vars_indep)
func_select_output_anova(lst_results)
lst_results
```
\pagebreak

#### Placental - Past or present malaria

```{r}

data <- df_2000_all %>% 
  filter(malaria_p_pastorprsnt == 1)

lst_results <- func_anova_multiple(data, vars_dep, vars_indep)
func_select_output_anova(lst_results)
lst_results
```
\pagebreak


# Anova with interaction MiRNA146 * clinical manifestation 

## Functions
```{r}

# Function to execute ANOVA test on 1 categorical and 1 continuous variable:
func_anovainteract <- function(data, var_infect, var_dep, var_indep){


  df_summary <- data %>%
  select(!!var_dep, !!var_indep) %>%
  group_by_(var_indep) %>%
  summarize_all(funs(mean = mean(., na.rm = T),
                     median = median(., na.rm = T),
                     sd   = sd(., na.rm = T),
                     min  = min(., na.rm = T),
                     max  = max(., na.rm = T))) %>% 
  as.data.frame() %>%
  kable()

  # pass character objects to lm() as formula:
  form <- paste(var_dep, "~", var_indep, "*", var_infect) %>% as.formula
  # run test, put NULL when test is not possible:
  model <- tryCatch(lm(form, data = data), error = function(e){ NULL })
  
  
  # store results:
  if(is.null(model)){
    p_value <- NA
  }else{
    p_value <- anova(model)$`Pr(>F)`[1:3]
    vars    <- rownames(anova(model))[1:3]
  } 
  
  results_anova <- data.frame(variables = vars, p = p_value)
  
  lst <- list("var_dep" = var_dep, "var_indep" = var_indep,
              "summary" = df_summary, 
              "results_anova" = results_anova)
  
  return(lst)
  
}

# Function to make parameter grid of al variable combinations to test, 
# then store results in list:
func_anovainteract_multiple <- function(data, var_infect, vars_dep, vars_indep){
  
  df_vars_combi <- 
    expand.grid(var_indep = vars_indep, var_dep = vars_dep) %>%
    mutate_all(as.character) # factors were behaving strange in the loop...
  
  lst <- list()
  
  for(i in 1:nrow(df_vars_combi)){
    
    var_dep_i   <- df_vars_combi$var_dep[i]
    var_indep_i <- df_vars_combi$var_indep[i]
    
    lst[[i]] <- func_anovainteract(data, var_infect, var_dep_i, var_indep_i)
  }  
  
  return(lst)
}

# Function to select output ANOVA model for quick overview:
func_select_output_anovainteract <- function(lst_results){
  
  df <- data.frame(matrix(ncol = 3, nrow = 0)) %>%
  setNames(c("clinical_manif", "variables", "P"))

  for(i in 1:length(lst_results)){
    
    var_dep   <- lst_results[[i]]$var_dep
    df_p      <- lst_results[[i]]$results_anova
    
    df_1test <- data.frame(df_p) %>%
      mutate(clinical_manif = rep(var_dep)) %>%
      select(clinical_manif, variables, P = p)
    
    df_border <- c("clinical_manif" = NA, "variables" = NA, "P" = NA)
    
    df <- rbind(df, df_1test) %>% rbind(df_border)
  }
  
  df <- df %>%
    kable()
  
  return(df)
}

```

## Agogo 1998

```{r}

## The variables to test:

vars_dep <- c("temp", "hb")
vars_indep <- c("mirna146", "mirna146_wt_vs_het", "mirna146_wt_vs_hom", 
                "mirna146_wt_vs_hetorhom", "mirna146_het_vs_hom")

expand.grid(var_indep = vars_indep, var_dep = vars_dep) %>%
  kable()
```
\pagebreak

### Multigravidae + primigravidae

```{r}

data <- df_1998_all 
var_infect <- "pcr_falc"

lst_results <- func_anovainteract_multiple(data, var_infect, vars_dep, vars_indep)
func_select_output_anovainteract(lst_results)
lst_results
```
\pagebreak

### Multigravidae only

```{r}

data <- df_1998_multi 
var_infect <- "pcr_falc"

lst_results <- func_anovainteract_multiple(data, var_infect, vars_dep, vars_indep)
func_select_output_anovainteract(lst_results)
lst_results
```
\pagebreak

### Primigravidae only

```{r}

data <- df_1998_prim
var_infect <- "pcr_falc"

lst_results <- func_anovainteract_multiple(data, var_infect, vars_dep, vars_indep)
func_select_output_anovainteract(lst_results)
lst_results
```
\pagebreak

## Agogo 2000

```{r}

## The variables to test:

vars_dep <- c("temp", "hb")
vars_indep <- c("mirna146", "mirna146_wt_vs_het", "mirna146_wt_vs_hom", 
                "mirna146_wt_vs_hetorhom", "mirna146_het_vs_hom")

expand.grid(var_indep = vars_indep, var_dep = vars_dep) %>%
  kable()
```
\pagebreak

### Peripherial - PCR

```{r}

data <- df_2000_all
var_infect <- "pcr_falc_m"

lst_results <- func_anovainteract_multiple(data, var_infect, vars_dep, vars_indep)
func_select_output_anovainteract(lst_results)
lst_results
```
\pagebreak

### Placental - PCR

```{r}

data <- df_2000_all
var_infect <- "pcr_falc_p"

lst_results <- func_anovainteract_multiple(data, var_infect, vars_dep, vars_indep)
func_select_output_anovainteract(lst_results)
lst_results
```
\pagebreak

### Placental - Past or present malaria

```{r}

data <- df_2000_all
var_infect <- "malaria_p_pastorprsnt"

lst_results <- func_anovainteract_multiple(data, var_infect, vars_dep, vars_indep)
func_select_output_anovainteract(lst_results)
lst_results
```
\pagebreak